File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int maxx = INT_MIN;
       int minn = prices[0];
       for(int i = 1; i < prices.size(); i++){
          int check = prices[i] - minn;
          maxx = max(maxx, check);
          minn = min(minn, prices[i]);
       } 
       if(maxx <= 0) return 0;
       return maxx;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    ans := 0
    maxx := prices[0]
    for i := 1; i < len(prices); i++{
        if ans < (prices[i]-maxx){
            ans = (prices[i]-maxx)
        }
        if prices[i] < maxx{
            maxx = prices[i]
        }
    }
    return ans
}